Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -147,12 +145,19 @@ def update(self):
 
         """
         
-        self.r = self.data - self.operator.direct(self.x)
+        # self.r = self.data - self.operator.direct(self.x)
+        self.operator.direct(self.x, out=self.r)
+        self.r.sapyb(-1, self.data, 1.0, out=self.r)
         
-        self.x += self.relax_par * (self.D*self.operator.adjoint(self.M*self.r))
+        # self.x += self.relax_par * (self.D*self.operator.adjoint(self.M*self.r))
+        self.r *= self.M
```

Comment:
I think this could happen in a single `sapyb`

```self.r.sapyb(-self.M, self.data, self.M, out=self.r)```
However since we need `-self.M` there needs to be another loop :(

Reply:
So does something need doing here or just meant as a comment?

Reply:
It is a comment. We could add to `sapyb`  `sign` parameters for `a` and `b` which would allow to do what I write without another for loop.